pkg/gardenlet/controller/backupbucket/reconciler.go

```diff
@@ -0,0 +1,457 @@
+		return reconcile.Result{}, fmt.Errorf("could not update status after reconciliation start: %w", updateErr)
+	}
+
+	secret, err := kutil.GetSecretByReference(ctx, r.GardenClient, &backupBucket.Spec.SecretRef)
+	if err != nil {
+		log.Error(err, "Failed to get backup secret", "secret", client.ObjectKey{Namespace: backupBucket.Spec.SecretRef.Namespace, Name: backupBucket.Spec.SecretRef.Name})
+		r.Recorder.Eventf(backupBucket, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, "Failed to get backup secret %s/%s: %w", backupBucket.Spec.SecretRef.Namespace, backupBucket.Spec.SecretRef.Name, err)
+		return reconcile.Result{}, err
+	}
+
+	if !controllerutil.ContainsFinalizer(secret, gardencorev1beta1.ExternalGardenerName) {
+		log.Info("Adding finalizer to secret", "secret", client.ObjectKeyFromObject(secret))
+		if err := controllerutils.AddFinalizers(ctx, r.GardenClient, secret, gardencorev1beta1.ExternalGardenerName); err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to add finalizer to secret: %w", err)
+		}
+	}
+
+	var (
+		reconcileExtensionBackupBucket = false
+		reconciliationSuccessful       = true
+
+		extensionSecret           = r.emptyExtensionSecret(backupBucket.Name)
+		extensionBackupBucketSpec = extensionsv1alpha1.BackupBucketSpec{
+			DefaultSpec: extensionsv1alpha1.DefaultSpec{
+				Type:           backupBucket.Spec.Provider.Type,
+				ProviderConfig: backupBucket.Spec.ProviderConfig,
+			},
+			Region: backupBucket.Spec.Provider.Region,
+			SecretRef: corev1.SecretReference{
+				Name:      extensionSecret.Name,
+				Namespace: extensionSecret.Namespace,
+			},
+		}
+	)
+
+	if err := r.SeedClient.Get(ctx, client.ObjectKeyFromObject(extensionSecret), extensionSecret); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+	} else {
+		// if the backupBucket secret data has changed, reconcile extension backupBucket
+		if !reflect.DeepEqual(extensionSecret.Data, secret.Data) {
+			reconcileExtensionBackupBucket = true
+		}
+	}
+
+	if err := r.reconcileExtensionBackupBucketSecret(ctx, backupBucket); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := r.SeedClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupBucket), extensionBackupBucket); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		// if the extension BackupBucket doesn't exist yet, create it
+		reconcileExtensionBackupBucket = true
+	} else if !reflect.DeepEqual(extensionBackupBucket.Spec, extensionBackupBucketSpec) { /* if the extensionBackupBucketSpec has changed, reconcile it*/
+		reconcileExtensionBackupBucket = true
+	} else {
+		// check for errors, and if none are present, sync generated Secret to garden
+		if extensionBackupBucket.Status.LastError != nil || extensionBackupBucket.Status.LastOperation.State != gardencorev1beta1.LastOperationStateSucceeded {
```

Comment:
As said before, we should only trigger a new reconciliation when we are in a final state, so besides checking for `Succeeded` you should also check for `Failed`. When it is `Error` or `Processing` then we should not need to trigger a new reconciliation (as per contract, it should be retried automatically by the extension controller).